src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterRepository.java

```diff
@@ -25,7 +25,7 @@
         @Param("before") Timestamp before
     );
 
-    Optional<Letter> findByMessageIdAndStatusOrderByCreatedAtDesc(String messageId, LetterStatus status);
+    Optional<Letter> findByChecksumAndStatusOrderByCreatedAtDesc(String checksum, LetterStatus status);
```

Comment:
This change means that currently deployed version searches for letters by message ID. This PR starts populating checksums only, which means that the old version will not be able to find letters created by the new version. This change in the repository should be done first.

Reply:
done